src/libaktualizr/uptane/fetcher.cc

```diff
@@ -79,21 +80,22 @@ bool Fetcher::fetchVerifyTarget(const Target& target) {
       if (!target.MatchWith(Hash(ds.hash_type, ds.hasher().getHexDigest()))) {
         throw TargetHashMismatch(target.filename());
       }
+      result = true;
     } else {
 #ifdef BUILD_OSTREE
       KeyManager keys(storage, config.keymanagerConfig());
       keys.loadKeys();
-      OstreeManager::pull(config.pacman.sysroot, config.pacman.ostree_server, keys, target, events_channel);
+      data::InstallOutcome outcome =
+          OstreeManager::pull(config.pacman.sysroot, config.pacman.ostree_server, keys, target, events_channel);
+      result = (outcome.first == data::UpdateResultCode::kOk);
```

Comment:
pull might also return kAlreadyProcessed, which shouldn't be considered as failure.

Reply:
Good call, fixed. Thanks!